Gems/Terrain/Code/Source/TerrainRenderer/TerrainMeshManager.cpp

```diff
@@ -662,18 +748,27 @@ namespace Terrain
 
         m_indexBuffer = CreateMeshBufferInstance(
             AZ::RHI::GetFormatSize(AZ::RHI::Format::R16_UINT),
-            aznumeric_cast<uint32_t>(m_indices.size()),
-            m_indices.data());
-
-        // Create x/y positions. These are the same for all sectors since they're in local space.
+            aznumeric_cast<uint32_t>(indices.size()),
+            indices.data());
 
-        struct VertexPosition
+        if (m_rayTracingEnabled)
         {
-            uint8_t m_posx;
-            uint8_t m_posy;
-        };
+            // Generate a 32 bit index buffer for ray tracing by copying and transforming the 16 bit index buffer.
```

Comment:
16-bit RayTracing index buffers are supported, so it should be possible to reuse the index buffer.